src/FakeItEasy/Core/EventCallHandler.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace FakeItEasy.Core
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    internal class EventCallHandler

```

Comment:
This contains most of the logic that was in `EventRule`, so that it can be reused in `Manage.Event...`